src/DeliveryCartReferencedContent.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+
+namespace Drupal\delivery;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\delivery\Entity\MenuLinkContent;
+use Drupal\Core\Entity\EntityInterface;
+
+class DeliveryCartReferencedContent {
+
+  static $count = 0;
+
+
+  /**
+   * Looks for any referenced menu items and adds them to the cart.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *
+   * @return false|int
+   */
+  public static function addMenuItems(EntityInterface $entity){
+    if(!self::isNode($entity)) {
```

Comment:
I agree that in most of the cases menu items are attached to nodes and not other entities, but we could also implement it in an generic way and don't care if the entity is a node. Then we can get the id and the entity type id from the entity object itself.